src/tmpfiles/tmpfiles.c

```diff
@@ -2633,6 +2750,50 @@ static int find_gid(const char *group, gid_t *ret_gid, Hashmap **cache) {
         return name_to_gid_offline(arg_root, group, ret_gid, cache);
 }
 
+static int parse_age_by_from_arg(const char *age_by_str, AgeBy *ret_age_by) {
+        const char *s = NULL;
+        AgeBy ab = 0;
+
+        static const struct {
+                char age_by_chr;
+                AgeBy age_by_flag;
+        } age_by_types[] = {
+                { 'a', AGE_BY_ATIME_F },
+                { 'b', AGE_BY_BTIME_F },
+                { 'c', AGE_BY_CTIME_F },
+                { 'm', AGE_BY_MTIME_F },
+                { 'A', AGE_BY_ATIME_D },
+                { 'B', AGE_BY_BTIME_D },
+                { 'M', AGE_BY_MTIME_D },
+        };
+
+        assert(age_by_str);
+        assert(ret_age_by);
+
+        s = age_by_str;
+
+        if (isempty(s))
+                return -EINVAL;
+
+        do {
+                /* Allow valid characters (whitespaces are ignored). */
+                if (!IN_SET(*s, 'a', 'b', 'c', 'm', 'A', 'B', 'M', ' ', '\t'))
```

Comment:
C should be allowed too. I mean, it doesn't make much sense, but I'd allow it anyway

Reply:
i am not a fan of the fact that the 8 chars basically appear twice here, once in thise IN_SET() expression, and once in the able above. I'd just have them in th table, and then abort use the for loop below as validation loop: when no entry is found in the table then consider the string invalid

Reply:
Hmm, I can probably add an extra argument `bool for_file`, and check for lower or upper case when calling `parse_age_by_from_arg()` for each of `age_by_file` and `age_by_directory`.